openfisca_us_data/datasets/ce/ce.py

```diff
@@ -0,0 +1,279 @@
+import numpy as np
+
+import h5py
+from pandas import DataFrame, concat
+
+from openfisca_us_data.utils import US, dataset
+from openfisca_us_data.datasets.ce.raw_ce import RawCE
+
+
+@dataset
+class CE:
+    name = "ce"
+    model = US
+    # Ratio of mean cash contributions to charity vs total cash contribs, 2016
+    # www.bls.gov/opub/btn/volume-8/the-relationship-between-cash-contributions-pretax-income-and-age.htm
+    proportion_cash_contrib_to_charity = 396.59 / 2080.85
+
+    def generate(year: int) -> None:
+        """Saves organized Consumer Expenditure data in HDF5 format via h5py
+
+        Computes "months in scope" from existing variables, and uses it to
+        adjust weights and compute annual estimates. Finally, those estimates
+        to compute estimated carbon emissions per household.
+
+        Args:
+            year (int): The year of the Consumer Expenditure to use.
+        """
+        year = int(year)
+        if year not in RawCE.years:
+            RawCE.generate(year)
+
+        raw_data = RawCE.load(year)
+        ce = h5py.File(CE.file(year), mode="w")
+        ce.create_group("/household")  # Household quarterly data
+        ce.create_group("/annual")  # Annual estimates
+
+        # Concatenate 5 "quarters" of fmli data, add months in scope ----------
+        df_list = []
+        for quarter_data in raw_data.keys():
+            df_list.append(raw_data[quarter_data])
+
+        fmli_df = concat(df_list)
+        mis = fmli_df.apply(
+            lambda row: months_in_scope(row["interview_mo"], row["nominal_quarter"]),
+            axis=1,
+        )
+        fmli_df.insert(7, "months_in_scope", mis)
```

Comment:
I see that the 7 follows from other `insert` statements for other column indexes after reviewing to the bottom. Unless the column order is important, I think the standard `df["col"] = vals` would be clearer.

Reply:
I agree here

Reply:
done. Also, after realizing that *raw.py uses PyTables, which is good at storing DataFrames in HDF5, I wondered if this logic would be better there. Let me know if you want me to move it.